Tests/SourceKitLSPTests/CodeActionTests.swift

```diff
@@ -1006,6 +1006,64 @@ final class CodeActionTests: XCTestCase {
     }
   }
 
+  func testConvertStringConcatenationToStringInterpolation() async throws {
+    try await assertCodeActions(
+      """
+      1️⃣#"["# + 2️⃣key + ": \\(3️⃣d) " + 4️⃣value + ##"]"##5️⃣
+      """,
+      ranges: [("1️⃣", "2️⃣"), ("3️⃣", "4️⃣"), ("1️⃣", "5️⃣")],
+      exhaustive: false
+    ) { uri, positions in
+      [
+        CodeAction(
+          title: "Convert String Concatenation to String Interpolation",
+          kind: .refactorInline,
+          edit: WorkspaceEdit(
+            changes: [
+              uri: [
+                TextEdit(
+                  range: positions["1️⃣"]..<positions["5️⃣"],
+                  newText: """
+                    ##"[\\##(key): \\##(d) \\##(value)]"##
+                    """
+                )
+              ]
+            ]
+          )
+        )
+      ]
+    }
+  }
+
+  func testConvertStringConcatenationToStringInterpolationNotShowUp() async throws {
+    //    ""
+    try await assertCodeActions(
+      """
+      1️⃣##"[\\##(2️⃣key): \\##(3️⃣d) 4️⃣\\##(value)]"##5️⃣
+      """,
+      ranges: [("1️⃣", "2️⃣"), ("3️⃣", "4️⃣"), ("1️⃣", "5️⃣")]
+    ) { uri, positions in
+      [
+        CodeAction(
+          title: "Convert string literal to minimal number of \'#\'s",
+          kind: .refactorInline,
+          edit: WorkspaceEdit(
+            changes: [
+              uri: [
+                TextEdit(
+                  range: positions["1️⃣"]..<positions["5️⃣"],
+                  newText: """
+                    ###"[\\##(key): \\##(d) \\##(value)]"###
+                    """
+                )
+              ]
+            ]
+          )
+        )
+      ]
+    }
+  }
+
```

Comment:
Could you also add a test case for multi-line string literals? I’m fine with it either not showing up the code action (I think string concatenations with multi-line string literals are less common than with single-line string literals) or it working properly. I just want to make sure that multi-line string literals don’t produce an invalid refactoring result.